compiler/semmagic.nim

```diff
@@ -500,6 +500,9 @@ proc semNewFinalize(c: PContext; n: PNode): PNode =
           getAttachedOp(c.graph, t, attachedDestructor).owner == fin:
         discard "already turned this one into a finalizer"
       else:
+        if fin.instantiatedFrom != nil and fin.instantiatedFrom != fin.owner: #undo move
+          fin.owner = fin.instantiatedFrom
+          pushOwner(c, fin)
```

Comment:
What? Without a corresponding `popOwner`?

Reply:
My bad. Dindt know `pop` existed, I thought the list a way to keep track of the owners for something else. This is an "undo". Will get rid of `push`